File: woodroid/src/main/java/me/gilo/woodroid/repo/order/RefundRepository.java
package me.gilo.woodroid.repo.order;

import me.gilo.woodroid.data.api.RefundAPI;
import me.gilo.woodroid.models.Order;
import me.gilo.woodroid.models.Refund;
import me.gilo.woodroid.models.filters.RefundFilter;
import me.gilo.woodroid.repo.WooRepository;
import retrofit2.Call;

import java.util.List;

public class RefundRepository extends WooRepository {

    private final RefundAPI apiService;

    public RefundRepository(String baseUrl, String consumerKey, String consumerSecret) {
       super(baseUrl, consumerKey, consumerSecret);
       apiService = retrofit.create(RefundAPI.class);

    }

    public Call<Refund> create(Order order, Refund refund) {
        return apiService.create(order.getId(), refund);
    }

    public Call<Refund> refund(Order order, int id) {
        return apiService.view(order.getId(), id);
    }

    public Call<List<Refund>> refunds(Order order) {
        return apiService.list(order.getId());
    }

    public Call<List<Refund>> refunds(Order order, RefundFilter refundFilter) {
        return apiService.filter(order.getId(), refundFilter.getFilters());
    }

    public Call<Refund> delete(Order order, int id) {
        return apiService.delete(order.getId(), id);
    }

    public Call<Refund> delete(Order order, int id, boolean force) {
        return apiService.delete(order.getId(), id, force);
    }


}

File: woodroid/src/main/java/me/gilo/woodroid/repo/order/RefundRepository.kt
package me.gilo.woodroid.repo.order

import me.gilo.woodroid.data.api.RefundAPI
import me.gilo.woodroid.models.Order
import me.gilo.woodroid.models.Refund
import me.gilo.woodroid.models.filters.RefundFilter
import me.gilo.woodroid.repo.WooRepository
import retrofit2.Call

class RefundRepository(baseUrl: String, consumerKey: String, consumerSecret: String) :
    WooRepository(baseUrl, consumerKey, consumerSecret) {

    private val apiService: RefundAPI = retrofit.create(RefundAPI::class.java)

    fun create(order: Order, refund: Refund): Call<Refund> {
        return apiService.create(order.id, refund)
    }

    fun refund(order: Order, id: Int): Call<Refund> {
        return apiService.view(order.id, id)
    }

    fun refunds(order: Order): Call<List<Refund>> {
        return apiService.list(order.id)
    }

    fun refunds(order: Order, refundFilter: RefundFilter): Call<List<Refund>> {
        return apiService.filter(order.id, refundFilter.filters)
    }

    fun delete(order: Order, id: Int): Call<Refund> {
        return apiService.delete(order.id, id)
    }

    fun delete(order: Order, id: Int, force: Boolean): Call<Refund> {
        return apiService.delete(order.id, id, force)
    }


}
